cdap-validation/bin/test_helpers.py

```diff
@@ -0,0 +1,71 @@
+import os
+import base64
+import urllib2
+import json
+import sys
+
+def onetime_auth(host,info):
+    # this should work with all, but when authentication is not set up correctly, this fails (http 403)
+    passman = urllib2.HTTPPasswordMgrWithDefaultRealm() # this creates a password manager
+    passman.add_password(None, host, info['username'], info['password'])
+    authhandler = urllib2.HTTPBasicAuthHandler(passman) # create the AuthHandler
+    opener = urllib2.build_opener(authhandler)
+    urllib2.install_opener(opener)
+    return urllib2
+
+def run_request(host,info):
+    user = info['username']
+    passwd = info['password']
+    try:
+        handle = urllib2.urlopen(host)
+        return handle
+    except: #except HTTPError, h: ### would like to do something like this
+        # try a different way (encoding headers)
+        req = urllib2.Request(host)
+        base64string = base64.encodestring('%s:%s' % (user, passwd))[:-1]
+        authheader =  "Basic %s" % base64string
+        req.add_header("Authorization", authheader)
+        try:
+            handle = urllib2.urlopen(req)
+            return handle
+        except IOError, e:
+            # here we shouldn't fail if the username/password is right
+            print "It looks like the username or password is wrong."
+            return 'noapi'
+
+def write_config_file(config, dir, file):
+    path = dir + file 
+    f = open(path, 'w')
+    f.write(config)
+    f.close()
+
+# needed for CM and other configs that require config API calls based on lists of services, etc.. 
+#    retrieved through another API call, e.g. 
+def convert_types_to_list(types):
+    data = json.loads(types)
+    services = []
+    upper_services = [ item for item in data["items" ] ]
+    for service in upper_services:
+        services.append(service.lower())
+    services.sort()
+    return services
+
+def get_config_and_write(url, subdir, file, cluster_info):
+    # run api config retrieval commands and write to individual files
+    user = cluster_info['username']
+    passwd = cluster_info['password']
+    config_in = run_request(url, cluster_info)
+    config = config_in.read()
+    write_config_file(config, subdir, file)
+
+def safe_get_config_and_write(url, user, passwd, subdir, file):
```

Comment:
seems like you'd never want the unsafe one... ie, you should just have one version of the method which handles errors appropriately.